snapcraft/internal/indicators.py

```diff
@@ -36,16 +36,22 @@ def download_requests_stream(request_stream, destination, message=None):
     if not request_stream.headers.get('Content-Encoding', ''):
         total_length = int(request_stream.headers.get('Content-Length', '0'))
 
-    if total_length:
-        progress_bar = ProgressBar(
-            widgets=[message,
-                     Bar(marker='=', left='[', right=']'),
-                     ' ', Percentage()],
-            maxval=total_length)
+    if total_length and is_dumb_terminal():
+        widgets = [message]
+        maxval = total_length
+    elif total_length and not is_dumb_terminal():
+        widgets = [message,
+                   Bar(marker='=', left='[', right=']'),
+                   ' ', Percentage()]
+        maxval = total_length
+    elif not total_length and is_dumb_terminal():
+        widgets = [message]
+        maxval = UnknownLength
     else:
-        progress_bar = ProgressBar(
-            widgets=[message, AnimatedMarker()],
-            maxval=UnknownLength)
+        widgets = [message, AnimatedMarker()]
+        maxval = UnknownLength
```

Comment:
The reason I did it like this is because I went to a talk a while back about presenting these as tables of truth for much faster view of the possible combinations.